core-blocks/embed/index.js

```diff
@@ -46,6 +40,224 @@ const findBlock = ( url ) => {
 	return 'core/embed';
 };
 
+export function getEmbedEdit( title, icon ) {
+	return class extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.switchBackToURLInput = this.switchBackToURLInput.bind( this );
+			this.setUrl = this.setUrl.bind( this );
+			this.processPreview = this.processPreview.bind( this );
+
+			this.state = {
+				html: this.props.preview ? this.props.preview.html : '',
```

Comment:
Can't we rely on the props directly and avoid all these state values now that the fetching is done in a HoC?

Reply:
I hadn't noticed, but yes, we can get rid of a _lot_ of the state!

Reply:
We can easily get rid of everything except `fetching` and `url`! Which is nice :)